packages/api/test/integration.test.ts

```diff
@@ -100,37 +101,93 @@ describe('integration tests', function () {
   });
 
   describe('header handling', function () {
-    it('should support supplying an `accept` header', async function () {
-      fetchMock.post('http://petstore.swagger.io/v2/pet', mockResponse.all);
+    describe('`authorization`', function () {
+      it('should support supplying an `authorization` header', async function () {
+        fetchMock.post('http://petstore.swagger.io/v2/pet', mockResponse.all);
 
-      const body = {
-        id: 1234,
-        name: 'buster',
-      };
+        const body = {
+          id: 1234,
+          name: 'buster',
+        };
 
-      const metadata = {
-        accept: 'text/xml',
-      };
+        const metadata = {
+          authorization: 'bearer 12345',
+        };
 
-      const { data } = await api(petstore as unknown as OASDocument).addPet(body, metadata);
-      expect(data.uri).to.equal('/v2/pet');
-      expect(data.requestBody).to.equal('{"id":1234,"name":"buster"}');
-      expect(data.headers).to.have.deep.property('accept', 'text/xml');
-      expect(data.headers).to.have.a.customUserAgent;
+        const sdk = api(petstore as unknown as OASDocument);
+
+        sdk.auth('buster');
```

Comment:
Should we throw a warning if the user passes both an `.auth` parameter as well as an `Authorization` header?

Reply:
We don't have a great way to throw any warnings outside of `console.*` and I wouldn't want to clutter up any integrations that come from this that aren't outright errors.

Reply:
Especially because `sdk.auth()` can't be chained, maybe you have `sdk.auth()` somewhere up your stack but where you need to make a single call you need/want to specify an auth header.